.github/e2e-tests-olm/action.yaml

```diff
@@ -75,6 +79,46 @@ runs:
           --namespace operators \
           --skip-tls
 
+        # NOTE: operator will deploy other pods as well which we don't want to wait for
+        kubectl wait --for=condition=Ready pods \
+          -l app.kubernetes.io/name=monitoring-stack-operator,app.kubernetes.io/component=operator \
+          --namespace=operators --timeout=300s
+
+    - name: Expose operator metrics endpoint
+      shell: bash
+      run: |
+        kubectl apply -n operators -f hack/kind/operator-metrics-service.yaml
+        sleep 2
+
+        # wait for the service to create an endpoint
+        while !kubectl get ep operator-metrics-service && [[ "$i" -le 5 ]]; do
+            (( i++ ))
+            echo " - $i ... repeat "
+            sleep 2
+        done
+
+    - name: Print operator reconcile errors before running tests
+      shell: bash
+      run: |
+        ./tmp/bin/promq -t http://localhost:30001/metrics  \
+          -q 'controller_runtime_reconcile_errors_total'
+
     - name: Run tests
       shell: bash
       run: go test -v ./test/e2e/...
+
+    - name: Print operator reconcile errors
+      shell: bash
+      run: |
+        ./tmp/bin/promq -t http://localhost:30001/metrics  \
+          -q 'controller_runtime_reconcile_errors_total'
+
+    - name: Ensure there are reconcile errors
```

Comment:
Should we make this assertion part of the query? E.g.
```
promq -t http://localhost:30001/metrics  -q 'controller_runtime_reconcile_errors_total == 0'
```
and then we just need to check that we get some results back.

Reply:
Good idea .. let me try :)

Reply:
I tried this but ... it gives us false negatives 

Given: 

```yaml
❯ ./tmp/bin/promq -t http://localhost:30001/metrics -q 'controller_runtime_reconcile_errors_total' -oyaml
- point:
    t: 1637113651012
    v: 6
  metric:
    __name__: controller_runtime_reconcile_errors_total
    controller: grafana-operator
    instance: http://localhost:30001/metrics
- point:
    t: 1637113651012
    v: 0
  metric:
    __name__: controller_runtime_reconcile_errors_total
    controller: monitoringstack
    instance: http://localhost:30001/metrics
```

```yaml
❯  ./tmp/bin/promq -t http://localhost:30001/metrics -oyaml -q 'controller_runtime_reconcile_errors_total == 0'
- point:
    t: 1637113544080
    v: 0
  metric:
    __name__: controller_runtime_reconcile_errors_total
    controller: monitoringstack
    instance: http://localhost:30001/metrics
```
Here monitoring stack controller hasn't produced any reconciliation errors, so checking if we have a result isn't sufficient. I will use use `count(controller_runtime_reconcile_errors_total != 0)` to assert that it is `0`

Reply:
I will revert to using `sum` since the number of reconciliation errors make more sense IMHO